common/test/java/com/couchbase/lite/logging/LogTest.kt

```diff
@@ -547,33 +545,85 @@ class LogTest : BaseDbTest() {
         Assert.assertTrue(msg.contains("bork"))
     }
 
-    // Verify that we can handle non-ascii content.
-    @Ignore("Need a way to coax LiteCore into logging a non-ascii string")
+    // There was a bug that caused a call to the c4log_getDomain
+    // from within the callback to deadlock the thread.
     @Test
-    fun testNonASCII() {
-        val hebrew = "מזג האוויר נחמד היום" // The weather is nice today.
+    fun testRecursiveLogging() {
+        val startLatch = CountDownLatch(1)
+        val finishLatch = CountDownLatch(1)
+
+        val logSinks = assertNonNull(LogSinksImpl.getLogSinks())
+        logSinks.file = FileLogSink.Builder().setDirectory(scratchDirPath!!).build()
+
+        // If there is a file logger, an attempt to log a message from within the callback
+        // will call c4log_getDomain, which would deadlock the thread.
+        logSinks.c4Log.setCallbackInstrumentation { _: String?, _: Int, _: String? ->
+            startLatch.countDown()
+            Log.e(LogDomain.DATABASE, "BANG!!!")
```

Comment:
Not sure if using Log.e() here will cause the dead lock in this test. It depends on whether Log.e() causes the log callback to be called or causes `c4log_getDomain()` to be called or not. I'm pointing this out because the platform's log() method will call to `ConsoleLogSink` and `CustomLogSink` directly without going through LiteCore.

If instead of calling Log.e(), call `C4TestUtils.forceLog()` or `new Database("foo")`, it will definitely hang.